### PR TITLE
[MRG] add --num-results/-n to gather

### LINKED ISSUE #1042: allow for truncating sourmash gather at the command line
it would be nice to have a way to truncate gather searching to just e.g. the first match, with -n 1 or something similar.

### PR DESCRIPTION
**Pull request by ctb**
Provides `-n/--num-results` argument to sourmash gather.

Fixes #1042.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
This should be ready for review @erikyoung85 @bluegenes

**Comment by ctb** [comment]
updated & ready for review again :)

File: tests/test_sourmash.py [hunk 1]
```diff
@@ -2563,6 +2563,37 @@ def test_gather_metagenome():
                     'NC_011294.1 Salmonella enterica subsp...' in out))
 
 
+def test_gather_metagenome_num_results():
+    # set a threshold on the gather output
+    with utils.TempDirectory() as location:
```
Comment (erikyoung85) [inline tests/test_sourmash.py:2568]: I think you could use the `@utils.in_tempdir` decorator for this test function. Actually, a question of mine is how important are python decorators? Do they help the code run, or are they really just for readability purposes.

Comment (ctb) [reply tests/test_sourmash.py]: hah, good catch! thanks!

in this case, for readability purposes. but don't discount readability, it's one of the most important things in a code base!

**Pull request merged**